### PR TITLE
Nodejs6 debugging support

### PR DESCRIPTION
**Pull request by micholas**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Infrastructure (change which improves the lifecycle management (CI/CD, build, package, deploy, lint, etc) of the application, but does not change functionality.)
- [ ] Technical debt (change which improves the maintainability of the codebase, but does not change functionality)
- [ ] Testing (change which modifies or adds test coverage, but does not change functionality)
- [ ] Documentation (change which modifies or adds documentation, but does not change functionality)

## Description

Added support for debugging node6 lambdas

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/shared/codelens/typescriptCodeLensProvider.ts [hunk 1]
```diff
@@ -75,7 +75,7 @@ export function initialize({
             port: debugPort!,
             localRoot: samProjectCodeRoot,
             remoteRoot: '/var/task',
-            protocol: 'inspector',
+            protocol: params.runtime === 'nodejs6.10' ? 'legacy' : 'inspector',
```
Comment (bryceitoc9) [inline src/shared/codelens/typescriptCodeLensProvider.ts]: Also applies to versions older than 6.10

Comment (micholas) [reply src/shared/codelens/typescriptCodeLensProvider.ts]: I wan't thinking we were going to support end-of-life versions of node including node4 or earlier.

Comment (bryceitoc9) [reply src/shared/codelens/typescriptCodeLensProvider.ts]: If it shows up in our creation workflow, we implicitly support it. We've had some discussion around here about culling down the list to only "currently-supported" runtimes (https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html), but I'm not sure who the authority is on that--Lambda (where the code still runs), SAM (where templates can still be created), us, or someone else.

**Comment by micholas** [comment]
Documenting here what has been discussed in conversation. There are two points of view:

1. If a user attempts to attach a debugger and a modal error dialog occurs, we should not attach the debugger and rather leave the user to figure out what to do to attach it themselves.
2. If a user attempts to attach a debugger, we should always retry until either the debugger is attached (as requested) or max retries is reached.

This PR is written from the 2nd point of view, we should try our best to do what the user asked us to do. I've requested others to test but thus far I believe I am the only one to have tested it. For me this code always eventually successfully attaches the debugger for all runtimes I've tested which are Node6, Node8, Python 2.7, Python 3.6 and Python 3.7.

**Comment by micholas** [comment]
Skipping Node6 support to to impending Node6 end of life.

**Pull request closed without merging**